---
layout: post2
permalink: /epigenomic_data_analysis_2016/
title: Epigenomic Data Analysis 2016 Student Page
header1: Epigenomic Data Analysis 2016
header2: Workshop pages for students
image: CBW_Epigenome-data_icon.jpg
---

<ul id="navmenu">
  <li><a id="back_to_top">Contents</a>
     <ul class="sub1">
     <li><a href="#course_schedule">Course Schedule</a></li>
     <li><a href="#q_a_forum">Workshop Q/A Forum</a></li>
     <li><a href="#laptop_setup">Laptop Setup Instructions</a></li>
     <li><a href="#pre_readings">Pre-Workshop Readings</a></li>
     <li><a href="#pre_tutorials">Pre-Workshop Tutorials</a></li>
      <li><a href="#day1">Day 1</a>
         <ul class="sub2">  
           <li><a href="#welcome">Welcome</a></li>
           <li><a href="#module_1">Module 1</a></li>
           <li><a href="#module_2">Module 2</a></li>
        </ul>
      </li>
       <li><a href="#day_2">Day 2</a>
          <ul class="sub2">
             <li><a href="#module_3">Module 3</a></li>
             <li><a href="#module_4">Module 4</a></li>
           </ul>
       </li>
    </ul>
  </li>
</ul>  

<br>

###  Course Schedule  <a id="course_schedule"></a>

  <a href="http://bioinformatics-ca.github.io/epigenomic_data_analysis_schedule_2016/">Schedule for June 20 to June 21, 2016</a>

[&uarr;](#back_to_top)

###  Workshop Q/A Forum <a id="q_a_forum"></a>

  Post your workshop questions <a href="https://noteapp.com/Epigenomics2016">here</a>!
  
### Workshop Survey

We appreciate your feedback!  Please fill out [our survey](https://www.surveymonkey.com/r/6NVXVQT)!
  
### Class Picture

![Class photo](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/epigenomics/img/class_photo_small.png?raw=true)

[&uarr;](#back_to_top)

###  Laptop Setup Instructions <a id="laptop_setup"></a>

  Instructions to setup your laptop can be found <a href="https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/epigenomics/laptop_instructions.md">here</a>.

[&uarr;](#back_to_top)

###  Pre-workshop Readings <a id="pre_readings"></a>

[Introduction to High-Performance Computing](http://insidehpc.com/hpc-basic-training/what-is-hpc/)

[Introduction to Compute Canada](https://www.computecanada.ca/research-portal/accessing-resources/)

[Introduction to epigenomics](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3080743/)

[Additional reading on ChIP-Seq](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3053263/)

[Additional reading on Methylation analysis](http://www.ncbi.nlm.nih.gov/pubmed/22986265/) 

###  Pre-workshop Tutorials <a id="pre_tutorials"></a>

[UNIX Preparation tutorials](http://www.ee.surrey.ac.uk/Teaching/Unix/): Please complete tutorials #1-3 on UNIX 

[Unix cheat sheet](http://www.rain.org/~mkummel/unix.html)

[IGV Tutorial](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_IGV_lab_2016/): Review how to use IGV Genome Browser if you have not used this tool before.

[UCSC Genome Browser tutorial](http://www.ncbi.nlm.nih.gov/pubmed/18514479)

[FastQC Introduction and examples](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/)


[&uarr;](#back_to_top)

### YouTube Playlist for Recorded Lectures

<img src="https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/images/Nova-Video-300px.png?raw=true" width="42"> [Recorded Lectures' Playlist](https://www.youtube.com/playlist?list=PL3izGL6oi0S-CGCK0dPj0Mwt1XBWCCWcq)


***

## R Review Session

[Lecture](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/resources/RReview_slides.pdf)

[RScript to use for Review](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/raw/master/2016_workshops/epigenomics/BiCG_2016_Rreview_Code.R)

[RScript to view Results](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/resources/R_Review_Session_Code.ipynb)

##  Day 1 <a id="day_1"></a>

###  Welcome <a id="welcome"></a>

  *<font color="#827e9c">Ann Meyer</font>* 
<br>

[&uarr;](#back_to_top)

###  Module 1: Introduction to ChIP Sequencing and Analysis <a id="module_1"></a>

  *<font color="#827e9c">Martin Hirst</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-1-2016)
  
  <img src="https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/images/Nova-Video-300px.png?raw=true" width="42"> [Recorded Lecture](https://youtu.be/7g16OnETZf0)
  
  [Lecture on Compute Canada](https://bioinformatics.ca/epigenomics-module-1-hpc-2016)
  
  [Using the Guillimin HPC](http://bioinformatics-ca.github.io/epigenomic_data_analysis_hpc_2016/)

  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module1_lab_2016/)
  
  
[&uarr;](#back_to_top)

###  Module 2: ChIP-Seq Alignment, Peak Calling, and Visualization <a id="module_2"></a>

  *<font color="#827e9c">Misha Bilenky</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-2-2016)
  
  <img src="https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/images/Nova-Video-300px.png?raw=true" width="42"> [Recorded Lecture](https://youtu.be/n6e6rdQsMmw)
  
  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module2_lab_2016/)

[&uarr;](#back_to_top)

##  Day 2 <a id="day_2"></a>

###  Module 3: Introduction to WGBS and Analysis <a id="module_3"></a>

  *<font color="#827e9c">Guillaume Bourque</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-3-2016)
  
  <img src="https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/images/Nova-Video-300px.png?raw=true" width="42"> [Recorded Lecture](https://youtu.be/4TdFn5MrD90)
  
  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module3_lab_2016/)

[&uarr;](#back_to_top)


###  Module 4: Downstream Analyses and Integrative Tools <a id="module_4"></a>

  *<font color="#827e9c">David Bujold</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-4-2016)
  
  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module4_lab_2016/)

[&uarr;](#back_to_top)

***
