---
layout: workshop_main_2day
permalink: /epigenomics_2017
title: Epigenomic Data Analysis
header1: Workshop Pages for Students
header2: Epigenomic Data Analysis 2017
image: /site_images/CBW_Epigenome-data_icon.jpg
---

# Welcome <a id="welcome"></a>

Welcome to Infectious Disease Genomic Epidemiology.  

The course schedule can be found [here](https://bioinformaticsdotca.github.io/epidemiology_2017_schedule).

If you have questions or comments, visit the workshop [Q&A forum](https://noteapp.com/Epi2017).

***

# Pre-Workshop Materials <a id="preworkshop"></a>

## Laptop Setup Instructions

Instructions to setup your laptop can be found [here](https://github.com/bioinformatics-ca/bioinformatics-ca.github.io/blob/master/2016_workshops/epigenomics/laptop_instructions.md).

## Pre-workshop Readings

[Introduction to High-Performance Computing](http://insidehpc.com/hpc-basic-training/what-is-hpc/)

[Introduction to Compute Canada](https://www.computecanada.ca/research-portal/accessing-resources/)

[Introduction to epigenomics](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3080743/)

[Additional reading on ChIP-Seq](http://www.ncbi.nlm.nih.gov/pmc/articles/PMC3053263/)

[Additional reading on Methylation analysis](http://www.ncbi.nlm.nih.gov/pubmed/22986265/) 

##  Pre-workshop Tutorials

[UNIX Preparation tutorials](http://www.ee.surrey.ac.uk/Teaching/Unix/): Please complete tutorials #1-3 on UNIX 

[Unix cheat sheet](http://www.rain.org/~mkummel/unix.html)

[IGV Tutorial](http://bioinformatics-ca.github.io/bioinformatics_for_cancer_genomics_IGV_lab_2016/): Review how to use IGV Genome Browser if you have not used this tool before.

[UCSC Genome Browser tutorial](http://www.ncbi.nlm.nih.gov/pubmed/18514479)

[FastQC Introduction and examples](http://www.bioinformatics.babraham.ac.uk/projects/fastqc/Help/)

***

# Day 1 <a id="day1"></a>

##  Welcome <a id="welcome"></a>

  *<font color="#827e9c">Ann Meyer</font>* 
<br>

##  Module 1: Introduction to ChIP Sequencing and Analysis <a id="module_1"></a>

  *<font color="#827e9c">Martin Hirst</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-1-2016)
  
  [Lecture on Compute Canada](https://bioinformatics.ca/epigenomics-module-1-hpc-2016)
  
  [Using the Guillimin HPC](http://bioinformatics-ca.github.io/epigenomic_data_analysis_hpc_2016/)

  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module1_lab_2016/)
  
  
##  Module 2: ChIP-Seq Alignment, Peak Calling, and Visualization <a id="module_2"></a>

  *<font color="#827e9c">Misha Bilenky</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-2-2016)
  
  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module2_lab_2016/)

## Integrated Assignment


# Day 2 <a id="day2"></a>

##  Module 3: Introduction to WGBS and Analysis <a id="module_3"></a>

  *<font color="#827e9c">Guillaume Bourque</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-3-2016)
  
  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module3_lab_2016/)


##  Module 4: Downstream Analyses and Integrative Tools <a id="module_4"></a>

  *<font color="#827e9c">David Bujold</font>*
  
  [Lecture](https://bioinformatics.ca/epigenomics-module-4-2016)
  
  [Lab practical](http://bioinformatics-ca.github.io/epigenomic_data_analysis_module4_lab_2016/)

***
